src/com/b44t/messenger/DcMsg.java

```diff
@@ -71,6 +73,21 @@ public boolean equals(Object other) {
         return this.getId()==that.getId() && this.getId()!=0;
     }
 
+    /**
+     * If given a message, calculates the position of the message in the chat
+     */
+    public static int getMessagePosition(DcMsg msg, ApplicationDcContext dcContext) {
+        int msgs[] = dcContext.getChatMsgs(msg.getChatId(), 0, 0);
+        int startingPosition = -1;
+        for(int i=0; i< msgs.length; i++ ) {
+            if(msgs[i] == msg.getId()) {
```

Comment:
i would tend to get `msg.getId()` out of the loop and not only rely on the optimizer. a chat with some 1000 messages may result in lots of ffi calls, done before the chat is displayed, on slow devices, this may be some 100 ms.